gpiozero/boards.py

```diff
@@ -1185,3 +1185,97 @@ def on(self):
+            'C': (True, False, False, True, True, True, False),
+            'D': (False, True, True, True, True, False, True),
+            'E': (True, False, False, True, True, True, True),
+            'F': (True, False, False, False, True, True, True),
+            'G': (True, False, True, True, True, True, False),
+            'H': (False, True, True, False, True, True, True),
+            'I': (False, False, False, False, True, True, False),
+            'J': (False, True, True, True, True, False, False),
+            'K': (True, False, True, False, True, True, True),
+            'L': (False, False, False, True, True, True, False),
+            'M': (True, False, True, False, True, False, False),
+            'N': (True, True, True, False, True, True, False),
+            'O': (True, True, True, True, True, True, False),
+            'P': (True, True, False, False, True, True, True),
+            'Q': (True, True, False, True, False, True, True),
+            'R': (True, True, False, False, True, True, False),
+            'S': (True, False, True, True, False, True, True),
+            'T': (False, False, False, True, True, True, True),
+            'U': (False, False, True, True, True, False, False),
+            'V': (False, True, True, True, True, True, False),
+            'W': (False, True, False, True, False, True, False),
+            'X': (False, True, True, False, True, True, True),
+            'Y': (False, True, True, True, False, True, True),
+            'Z': (True, True, False, True, True, False, True),
+            '-': (False, False, False, False, False, False, True),
+            ' ': (False, False, False, False, False, False, False),
+            '=': (False, False, False, True, False, False, True)
+        }
+        
+        super(SevenSegmentDisplay, self).__init__(*pins, pwm=pwm, active_high=active_high, initial_value=initial_value)
+    
+    def display(self, char):
+        char = str(char).upper()
+        if len(char) > 1:
+            raise ValueError('only a single character can be displayed')
+        if char not in self._layouts:
+            raise ValueError('there is no layout for character - %s' % char)
+        layout = self._layouts[char]
+        for led in range(7):
+            self[led].value = layout[led]
+            
+    def display_hex(self, hexnumber):
+        self.display(hex(hexnumber)[2:])
+
+    @property
+    def decimal_point(self):
+        # does the 7seg display have a decimal point (i.e pin 8)
+        if len(self) > 7:
+            return self[7].value 
+        else:
+            raise OutputDeviceError('there is no 8th pin for the decimal point')
+    
+    @decimal_point.setter
+    def decimal_point(self, value):
+        if len(self) > 7:
+            self[7].value = value
+        else:
+            raise OutputDeviceError('there is no 8th pin for the decimal point')
+            
+    def set_char_layout(self, char, layout):
+        char = str(char)
```

Comment:
Hmm, this should probably be `char = str(char).upper()` to match the `display` function?